src/commands/synthetics/__tests__/run-tests-lib.test.ts

```diff
@@ -85,47 +102,50 @@ describe('run-test', () => {
       ).rejects.toThrow(new CiError('NO_TESTS_TO_RUN'))
     })
 
-    test('should apply config override for tests triggered by public id', async () => {
-      const getTestsToTriggersMock = jest.spyOn(utils, 'getTestsToTrigger').mockReturnValue(
-        Promise.resolve({
-          initialSummary: utils.createInitialSummary(),
-          overriddenTestsToTrigger: [],
-          tests: [],
-        })
-      )
-      jest.spyOn(utils, 'runTests').mockImplementation()
+    test.each(compat)(
+      'should apply config override for tests triggered by public id ($compat)',
+      async ({defaultTestOverrides}) => {
+        const getTestsToTriggersMock = jest.spyOn(utils, 'getTestsToTrigger').mockReturnValue(
+          Promise.resolve({
+            initialSummary: utils.createInitialSummary(),
+            overriddenTestsToTrigger: [],
+            tests: [],
+          })
+        )
+        jest.spyOn(utils, 'runTests').mockImplementation()
 
-      const startUrl = '{{PROTOCOL}}//myhost{{PATHNAME}}{{PARAMS}}'
-      const locations = ['location1', 'location2']
-      const userConfigOverride = {locations, startUrl}
+        const startUrl = '{{PROTOCOL}}//myhost{{PATHNAME}}{{PARAMS}}'
+        const locations = ['location1', 'location2']
+        const userConfigOverride = {locations, startUrl}
 
-      const apiHelper = {}
+        const apiHelper = {}
 
-      jest.spyOn(api, 'getApiHelper').mockImplementation(() => ({} as any))
+        jest.spyOn(api, 'getApiHelper').mockImplementation(() => ({} as any))
 
-      await expect(
-        runTests.executeTests(mockReporter, {
-          ...ciConfig,
-          global: userConfigOverride,
-          publicIds: ['aaa-aaa-aaa', 'bbb-bbb-bbb'],
-        })
-      ).rejects.toThrow()
-      expect(getTestsToTriggersMock).toHaveBeenCalledWith(
-        apiHelper,
-        expect.arrayContaining([
-          expect.objectContaining({id: 'aaa-aaa-aaa', testOverrides: userConfigOverride}),
-          expect.objectContaining({id: 'bbb-bbb-bbb', testOverrides: userConfigOverride}),
-        ]),
-        expect.anything(),
-        false,
-        false,
-        false
-      )
-    })
+        await expect(
+          runTests.executeTests(mockReporter, {
+            ...ciConfig,
+            [defaultTestOverrides]: userConfigOverride,
+            publicIds: ['aaa-aaa-aaa', 'bbb-bbb-bbb'],
+          })
+        ).rejects.toThrow()
+        expect(getTestsToTriggersMock).toHaveBeenCalledWith(
+          apiHelper,
+          expect.arrayContaining([
+            expect.objectContaining({id: 'aaa-aaa-aaa', testOverrides: userConfigOverride}),
+            expect.objectContaining({id: 'bbb-bbb-bbb', testOverrides: userConfigOverride}),
```

Comment:
As you can see, we expect the object to contain `testOverrides` even in the deprecated case because of our backwards compatibility translation layer.